### PR TITLE
Add update strategy and node selector overlay to vsphere-csi 2.4.1

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

## What this PR does / why we need it

We need to enforce the update strategy of all the deployment and daemonset running on Supervisor cluster. Also, add nodeSelector configuration to deployments.

## Which issue(s) this PR fixes

Fixes: # https://github.com/vmware-tanzu/tanzu-framework/issues/1850

## Describe testing done for PR

Manually tested the templates

### CODE REVIEW

File: addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-strategy-overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+
+#! We are adding this overlay in the package to accomandate the need from vSphere supervisor cluster:
```
Comment (vijaykatam) [inline addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-strategy-overlay.yaml:4]: typo in accomandate

File: addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-strategy-overlay.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#! `deployment.spec.strategy.type` is configured to `RollingUpdate`
+#! `deployment.spec.strategy.rollingUpdate.maxUnavailable` is set to `0`.
```
Comment (vijaykatam) [inline addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-strategy-overlay.yaml:6]: nitpick: we aren't enforcing these numbers with overlays so the comments are confusing.

Comment (blc1996) [reply addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-strategy-overlay.yaml:6]: Comment states `#! This overlay makes configuring the above parameters possible`, will refactor the comments for all packages in a follow up MR.

**Review by vijaykatam — APPROVED** [review]
LGTM

**Pull request merged**